shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -1056,6 +1056,11 @@ public void onAttach(@NonNull Context context) {
     delegate.onAttach(context);
     if (getArguments().getBoolean(ARG_SHOULD_AUTOMATICALLY_HANDLE_ON_BACK_PRESSED, false)) {
       requireActivity().getOnBackPressedDispatcher().addCallback(this, onBackPressedCallback);
+      // By default, Android handles backs, and predictive back is enabled. This
+      // can be changed by calling setFrameworkHandlesBack. For example, the
+      // framework will call this automatically in a typical app when it has
+      // routes to pop.
+      onBackPressedCallback.setEnabled(false);
```

Comment:
You probably also need to update the implementation of `popSystemNavigator()` to restore the previous state of the callback-enablement, instead of unconditionally re-enabling it on line 1673.

Reply:
@math1man Thanks for jumping back in here, I've been trying to catch back up on your comments from the last PR. Pushing a fix for that now.

Reply:
No worries. Honestly, I think besides this comment everything looks good. Happy to look into anything specific you might be concerned about though.